### PR TITLE
Add Ecovacs binary sensors

### PR DESCRIPTION
**Pull request by edenhaus**
## Proposed change

SSIA

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/108544
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/ecovacs.markdown [hunk 1]
```diff
@@ -12,20 +12,28 @@ ha_codeowners:
...
 
 Additional note: There are some issues during the password encoding. Using some special characters (e.g., `-`) in your password does not work.
 
+## Provided entities
+
+The Ecovacs integration provides a vacuum {% term entity %} for each device that is connected to your Ecovacs account.
+
+Using the vacuum entity, you can monitor and control your Ecovacs Deebot vacuum.
+
+Additionally, **depending on your model** the integration provides the following entities:
```
Comment (c0ffeeca7) [inline source/_integrations/ecovacs.markdown]:
```suggestion
Additionally, **depending on your model**, the integration provides the following entities:
```

File: source/_integrations/ecovacs.markdown [hunk 2]
```diff
@@ -12,20 +12,28 @@ ha_codeowners:
...
+
+- **Binary sensor**: 
+  - `Mop attached`: On if the mop is attached. Some models reports a state change only if the state of the vacuum has changed.
```
Comment (c0ffeeca7) [inline source/_integrations/ecovacs.markdown]: I don't understand. You are saying a state change is reported only if the state has changed. Isn't that expected behavior? I'm probably missing something here.

Comment (edenhaus) [reply source/_integrations/ecovacs.markdown]: Some models report a state change of the mop attached sensor only when the vacuum entity's state has changed.

For example:
- Vacuum is docked with the mop attached
- You remove the mop (no update will be pushed to HA)
- You start a new cleaning so the state of the vacuum changes to `cleaning`, and also the binary sensor state update to no mop attached.
I assume that the robot has some sort of sleep mode, where he will not check if a mop is attached or not

File: source/_integrations/ecovacs.markdown [hunk 3]
```diff
@@ -12,20 +12,28 @@ ha_codeowners:
...
+## Provided entities
+
+The Ecovacs integration provides a vacuum {% term entity %} for each device that is connected to your Ecovacs account.
+
+Using the vacuum entity, you can monitor and control your Ecovacs Deebot vacuum.
+
+Additionally, **depending on your model**, the integration provides the following entities:
+
+- **Binary sensor**: 
+  - `Mop attached`: On if the mop is attached. Some models reports a state change only if the state of the vacuum has changed.
```
Comment (c0ffeeca7) [inline source/_integrations/ecovacs.markdown]:
```suggestion
  - `Mop attached`: On if the mop is attached. Note: If you do not see the state change to `Mop attached` in Home Assistant, you may need to wake up the robot in order to push the state change. Some models report an entity state change only if the overall status of the vacuum has changed. For example, if the overall state changes from `sleep` to `cleaning`. 
```
Thank you for the explanation! So maybe something along those lines?

**Review by c0ffeeca7 — APPROVED** [review]
Thank you, @edenhaus :+1:

**Pull request merged**